pkg/defaults/defaults.go

```diff
@@ -127,6 +138,13 @@ func fromConfig(
 	}
 	for _, rawStep := range rawSteps {
 		if testStep := rawStep.TestStepConfiguration; testStep != nil {
+			if testStep.ClusterClaim != nil {
+				if hiveClient == nil {
+					return nil, nil, fmt.Errorf("cannot claim a cluster without providing a Hive kubeconfig")
+				}
+				clusterClaimStep := steps.ClusterClaimStep(testStep.ClusterClaim, hiveClient, client, jobSpec)
+				buildSteps = append(buildSteps, clusterClaimStep)
+			}
```

Comment:
This is a subtle issue: this validation happens at configuration load time, before the step graph is pruned.  The result is this type of validation will fail even if the step is not targeted (via `--target`).

The `Validate` method in the `Step` interface was added for this, check the `LeaseStep` implementation.